core/runtime/stats.rs

```diff
@@ -27,10 +27,17 @@ impl RuntimeActivityStatsFactory {
         .resources
         .push((resource.0, resource.1.to_string()))
     }
-    let timers = TimerStats {
-      intervals: vec![],
-      timers: vec![],
+    let timer_count = self.context_state.timers.len();
+    let mut timers = TimerStats {
+      timers: Vec::with_capacity(timer_count),
+      repeats: BitSet::with_capacity(timer_count),
     };
+    for (timer_id, repeats) in &self.context_state.timers.iter() {
+      if repeats {
+        timers.repeats.insert(timers.timers.len());
+      }
```

Comment:
Could you explain this line?

Reply:
`repeats` is a bitset that reports whether a given index in the ID array is an interval (if true) or a timer (if false). I'll put this into a comment.